Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpProjectTestsExecutionStrategy.cs

```diff
@@ -267,5 +269,26 @@ protected virtual void ExtractTestNames(IEnumerable<TestContext> tests)
                 }
             }
         }
+
+        protected void EnsureAssemblyNameIsCorrect(Project project)
+        {
+            var assemblyNameProperty = project.AllEvaluatedProperties.FirstOrDefault(x => x.Name == "AssemblyName");
+            if (assemblyNameProperty == null)
+            {
+                throw new ArgumentException("Project file does not contain Assembly Name property!");
+            }
+
+            var csProjFullpath = project.FullPath;
+            string projectName = this.ExtractProjectNameFromCsProjfile(csProjFullpath);
+            project.SetProperty("AssemblyName", projectName);
+        }
+
+        private string ExtractProjectNameFromCsProjfile(string csprojFile)
+        {
+            var indexOfLastSlash = csprojFile.LastIndexOf("\\");
```

Comment:
Use Path.GetFilenameWithoutExtension instead, also there is no need for a separate method, just extract the project name in the EnsureAssemblyNameIsCorrect method, this one seems redundant.